### DISCUSSION TITLE
Use theme based on user/role

### DISCUSSION THREAD

**Discussion by m-pernkopf** [filamentphp/filament#3389] [Help]
Dear filament community,

I am currently struggling with figuring out how to load a specific css for specific roles or users.
Tailwind & vite is setup properly (overwrites work) for single theme but I cannot find a clue how to
compile multiple css files so that the corresponding css file will be loaded when user logs in.
It's a kind of backend branding for multiple customers/users (f.ex. user a: green theme, user b: grey theme, user c: all font-size increased, ...).

Can you please point out if a) this is possible in filamentphp and b) there is a piece of documentation or a poc as a starting point for further investigation/evaluation.

Best regards, Michael

**Comment by danharrin** [comment]
Before you call `Filament::registerTheme()`, check the role, and register a different theme?

**Reply by m-pernkopf** [reply]
Hello,

thank you for your input. 
I already tried and used the registerTheme Method inside my boot() of AppServiceProvider, but it seems that this function is no longer available in filamentphp 2.0 (I followed the guide here (https://filamentphp.com/docs/1.x/admin/theming#registering-a-theme) but found out, that this documentation is for v1.x.
In v2.x this propably has to be handled a slightly different way.

Regards, Michael

**Reply by danharrin** [reply]
https://filamentphp.com/docs/2.x/admin/appearance#building-themes

**Reply by m-pernkopf** [reply]
I found out that there seems to be an additional problem. AppServiceProdiver is loaded before middleware (where the user will be stored in the session), so the authenticated user seems not to be available here. It can somehow be done by calling a view and get userdata from there.

**Reply by danharrin** [reply]
Okay, add a middleware after authentication that registers the theme them

**Reply by m-pernkopf** [reply]
> https://filamentphp.com/docs/2.x/admin/appearance#building-themes

 That's exactly what I've tried out, don't know what changed since my last try, but now at least the registeredTheme css file is loaded.

Any hints on how to add a middlware after authentication that registers the theme?
